### DISCUSSION TITLE
Welcome to m2strm Discussions!

### DISCUSSION THREAD

**Discussion by trix7777** [trix7777/m2strm#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by 3v3rm4rt3lL** [comment]
Hello I new of this. I couldnt find the .exe file, how should I run it?, I've trying to compile in CS but it show me errors. If you could me explain. Thank you!!

**Reply by trix7777** [reply]
Hi! You need Visual Studio to compile it. You can find a pre compiled version of every release in the right column under Releases here: https://github.com/trix7777/m2strm
Direct link: https://github.com/trix7777/m2strm/releases
Then click "Assets" and you will see the exe.

**Reply by 3v3rm4rt3lL** [reply]
Thank you so much. it works perfect. 

Just  I have a question, how could I get my VOD Series saves in a category? for example, I have in my IPTV provider Series Categories like Disney, Marvel, but I want to keep this categories in folders, Like  VOD SERIES - DISNEY - 101 DALMATIANS.STRM , VOD SERIES - MARVEL - SPIDERMAN 3.STRM 

If you can help me could be great.
Have a Great Day!!

**Reply by trix7777** [reply]
That's kind of a specific IPTV provider function, I've never seen a provider dividing it up like that, and it's not even a "real" category. I wonder, to what end? What are you using watching these movies? Kodi or some other media player? I use Jellyfin for indexing and Kodi for watching. If my provider would name the movies like that (I assume you mean movies and not series, as your examples both are movies), I would be more prone to stripping the DISNEY and MARVEL stuff out from the name instead of keeping it. The IPTV provider would be much better off using the M3U tag #EXTGENRE instead.

**Comment by 3v3rm4rt3lL** [comment]
I'm using it for watch in Emby,
Sorry may I do not explain well. My IP Provider has categories in series as well in movies. I used a setting call it "MovieGroupSubdirEnabled" = TRUE 
and for movies it works perfect to me, because I can keep categories (Groups) in folders. But for me, could be great if I can use the same setting in Series. 
For example, my provider has series categories like:
"Netflix Series -> Counterpart -> Counterpart S01E01" 
"Amazon PRIME Series -> Warriors of Liberty City -> Warriors of Liberty City S01E05 "
"Action Series -> La Casa de Papel -> La casa de Papel S01E10 " 
---
MY IPTV PROVIDER M3U_PLUS EXAMPLE
#EXTINF:-1 tvg-id="" tvg-name="Naruto Shippuden S12 E02" tvg-logo="https://image.tmdb.org/t/p/w300/aUAI5P2JYksYcolkLlQ31I8nLRa.jpg" group-title="Anime",Naruto Shippuden S12 E02
http://abcd.com:8080/series/abcde/fghij/1178702.mp4
--
So I would like to keep series as groups. and not all series in just one folder.

by the way, do you have where I can donate you for a coffe?
Thank you.

**Reply by trix7777** [reply]
If you could please tell me the reason for wanting that behavior, I would maybe consider implementing the feature if I find the reason reasonable, and if it's doable with current code. If using Emby or Jellyfin, as we both do, there is no need for categorizing series like that as that categorization would be lost when Emby does the indexing anyway.

EDIT: I took a quick glance at the code and it seems pretty easy to implement it. You could try it yourself if you want, just change the code line:

```
//combine path for series
combinedTypeDir = Path.Combine(SeriesDir, seriesNAME);
```
to
```
//combine path for series
combinedTypeDir = Path.Combine(SeriesDir, GROUP, seriesNAME);
```
And it should output just like you want it to.

**Reply by 3v3rm4rt3lL** [reply]
This is absolutely great.... IT WORKS perfect. Thank so much.

Please, add where we can donate in the future. 
Have a nice day :).

**Reply by trix7777** [reply]
v3.0.2.2 now has the option for SeriesGroupSubdirEnabled.

**Comment by calexicoca** [comment]
I tried to create config file the best way I can. But when I ran it all I get is for channels and VOD bunch of "_NONAME.strm" I would really like to use this great program :(

**Reply by trix7777** [reply]
There is no way for me to know what is happening without looking at both the _NONAME.strm, logs and your original.m3u8. Just make sure you remove all private info in them before posting them here.

**Reply by calexicoca** [reply]
Thank you so much for willingness to help (a newbie). I removed personal info. 

I got a million files like the one I sent you (when I click on them, they play. All files in VOD and TV channels are with the same name only in NOGROUP folder (with number). I know I ask a lot but if you have a sample config file with wanted and not wanted groups.

[1 _NONAME.strm.txt](https://github.com/trix7777/m2strm/files/7722309/1._NONAME.strm.txt)
[m2strm.exe().config.txt](https://github.com/trix7777/m2strm/files/7722310/m2strm.exe.config.txt)
[m2strm_20211215.log](https://github.com/trix7777/m2strm/files/7722319/m2strm_20211215.log)
[dupes.log](https://github.com/trix7777/m2strm/files/7722320/dupes.log)
[newgroups.log](https://github.com/trix7777/m2strm/files/7722321/newgroups.log)
[allgroups.log](https://github.com/trix7777/m2strm/files/7722322/allgroups.log)
[new_202112151105.log](https://github.com/trix7777/m2strm/files/7722323/new_202112151105.log)
[uwgroups.log](https://github.com/trix7777/m2strm/files/7722327/uwgroups.log)

**Reply by trix7777** [reply]
I need the original.m3u8 as well. You can use [Notepad++](https://notepad-plus-plus.org/downloads/) to search and replace the username and password in it before uploading it here. I also suggest you replace the real base URL.
Furthermore I noticed that you are missing the UserPort in your m2strm.exe.config.
You can find the example that you ask for [here](https://github.com/trix7777/m2strm) under `Example content, with both wanted (//) and not wanted groups`

**Reply by calexicoca** [reply]
I uploaded my M3U file here https://www98.zippyshare.com/v/q92ls9gl/file.html

Did replacement as you suggested.

As far as examples I saw help file but still having difficulties. My IPTV list does not have UserPort (as you can see in the enclosed file.

**Reply by trix7777** [reply]
Example content of a uwgroups.cfg would be:
```
//_NOGROUP
Sweden
Norway
Denmark
Finland
//Series: English [Multi-Sub]
//VOD: Premiere Cinemas [Multi-Sub]
```

You UserPort seem to be 8000 from what I can see in your m3u8. However, that does not seem to be the problem in this case, but the problem is that your IPTV-provider does not provide any groups in the m3u8 which from what I can see isn't even a m3u8 file at all, just an ordinary m3u. Can you please let me know how that link that you have to your m3u8 look like? Remove sensitive information. Eg. http://ip.tv:8080/get.php?username=ABCDEFGHIJ&password=0123456789&type=m3u_plus&output=ts

**Reply by calexicoca** [reply]
This is the format:
http://site.com/get.php?username=xxxxxxxx&password=yyyyyyyyyy&type=m3u&output=mpegts

**Reply by calexicoca** [reply]
BTW how do i get a list of groups (I don't think I am able to edit it manually

**Reply by trix7777** [reply]
If you compare the example I gave with the one that you've posted above, you'll see that it differentiates mainly in two ways, your link is to type=m3u -- which m3u, my example is for type=m3u_plus -- which is the same as m3u8, the other is that the output, ts vs mpegts. I suggest you try to change your links according to my example, mainly change the type=m3u to type=m3u_plus so it would look like http://site.com/get.php?username=xxxxxxxx&password=yyyyyyyyyy&type=m3u_plus&output=mpegts

**Reply by calexicoca** [reply]
Thanks I will try and report back

**Reply by calexicoca** [reply]
Sorry for my ignorance but where I can put the adapted playlist address in config file. If it calls from keys URL, Username and password it will still be as before.

**Reply by calexicoca** [reply]
When I did change playlist address as you suggested, I got m3u8 playlist (with tvg-id; tvg-name; tvg-logo;  group-title). Is there a way i can put that into config file and disable URL; User and Password?

**Reply by calexicoca** [reply]
It works with playlist file!!! Created all as it should. I will try now with uwgroups.cfg

**Reply by trix7777** [reply]
From what I can see you did not use m2strm to download the m3u8 at all as you have DownloadM3U8Enabled set to false, which must mean that you've given m2strm a location of a already downloaded m3u8 by means of other way. You should have DownloadM3U8Enabled set to true and UserPort set to 80 (as in http). MovieGroupSubdirEnabled should be default set to false.

**Comment by calexicoca** [comment]
> MovieGroupSubdirEnabled should be default set to false.

Does this mean that VOD Movies and VOD series will not have subfolders for Movies and TV Shows? I would like to have that so I can scrape and import in Kodi

**Reply by trix7777** [reply]
For Kodi, Emby or other scraper there is no need for a group subdir for movies. In fact, in some situations,  you could get problems with the indexing if there was a group subdir for movies. I suggest you try instead of asking a question like this.

**Reply by trix7777** [reply]
That might have come out a bit harsh, sorry for that. I just mean it's probably faster to experiment yourself than waiting for me to reply =) I hope you get it to work the way you want it to.

**Reply by calexicoca** [reply]
No problem. I am sorry that I ask stupid questions. Thanks for all your help. Of course I try things but the results are not always ... successful.

**Comment by laurent734** [comment]
hello,
I have an error when uploading my m3u list.

C: Windows system32>C: Users laure Documents m2strm/m2strm.exe /M
*** Download M3U8 file to: C: Users laure Documents m2strm original.m3u8
28/12/2021 16:20:35: ERROR: Invalid URI: Unable to determine URI format

How do I fix my problem?
thank you for your work

**Reply by trix7777** [reply]
I'm not a magician that with my mind can read your log and configuration files. You've probably did not configure the m2strm.exe.config correctly or not at all. Try running it with /G instead and follow the guide =)

**Reply by laurent734** [reply]
sorry!
I have already modified the config file.

<?xml version="1.0" encoding="utf-8"?>
<configuration>
    <appSettings>
        <add key="BaseDirectory" value="C:\Users\laure\Documents\m2strm\" />
        <add key="OutDirectory" value="C:\Users\laure\Documents\m2strm\" />
        <add key="m3u8File" value="" />
        <add key="MoviesSubDir" value="VOD Movies" />
        <add key="SeriesSubDir" value="VOD Series" />
        <add key="TVSubDir" value="TV Channels" />
        <add key="DeletePreviousDirEnabled" value="False" />
        <add key="UnwantedCFGEnabled" value="True" />
        <add key="VerboseConsoleOutputEnabled" value="False" />
        <add key="ProgramLogEnabled" value="True" />
        <add key="PurgeFilesEnabled" value="True" />
        <add key="MovieGroupSubdirEnabled" value="False" />
        <add key="DownloadM3U8Enabled" value="True" />
        <add key="http://mat4k.com" value="" />
        <add key="80" value="" />
        <add key="4678966134" value="" />
        <add key="4678966134" value="" />
    </appSettings>
</configuration>

sorry about my English

**Reply by laurent734** [reply]
I can’t post config

**Reply by laurent734** [reply]
j ai trouvé ! :) merci

**Comment by laurent734** [comment]
hello,
thanks for your work! I went from m3utostrm to m2strm and is much faster..:)
I’m having a problem with the iptv provider.
I have characters that block scrap on kodi ( -FR - -N- -VM - and .MULTI ).
Do you have an option to override these characters?

**Reply by trix7777** [reply]
I have no idea what you are asking.

**Reply by laurent734** [reply]
hello,
sorry j use google for translation.
I have characters in strm filenames that I want to remove because it blocks the search on tmbd with kodi ..
ex:-FR - A Second Chance - Rivals! (2019)
-VM - Too good (2021)
3 Days to Kill (2014) (MULTi)

**Reply by laurent734** [reply]
[exemple m3u.txt](https://github.com/trix7777/m2strm/files/7794165/exemple.m3u.txt)

**Reply by trix7777** [reply]
Check out 3.0.2.1 [release](https://github.com/trix7777/m2strm/releases).

**Reply by laurent734** [reply]
HELLO,
I just tested and it works wonderfully.
thank you very much:):).
I have gone through all the files and there are still a few important words to delete.
I made a file with examples respecting: space, upper case, lower case and point. I have the impression that it is they who are causing the problem.

[UNWANTED WORDS.txt](https://github.com/trix7777/m2strm/files/7795822/UNWANTED.WORDS.txt)

thank you again for your work and happy new year.

**Reply by trix7777** [reply]
I am just about to update to 3.0.2.2 in about an hour or so, try that and then report back.
I need the original m3u8-file from you to fix that. Remove personal information from it before uploading.

**Reply by laurent734** [reply]
you can download it with its information ..
it is a free 24 hour test.
It ends at 7 p.m.
url&port=http:/!!!!!!.com:8080
user =46337126!!!!!!6961
pass=463371269!!!!!961
this is the complete version with films & series for all countries available.
I just suppressed the live tv
thank you

**Reply by trix7777** [reply]
Working on it now. The provider has made many many mistakes in the m3u8 which I will correct some of them but far from all. E.g. some titles starts with blank space and some are named `|FR|` and the other `| FR|` and a third `| FR |`.

**Reply by laurent734** [reply]
okay !
I tested the version v3.0.2.2.
there is only the .MULTI which persists.
ex: The Revenant.MULTi
there is no space between the two so he considers that to be the name of the movie.
for the FR N VM it's ok! all disappeared

**Reply by trix7777** [reply]
v3.0.2.3 released. Should work with your provider now.

**Reply by laurent734** [reply]
Hello !
I tested v3.0.2.3.
It's just magic, you have to solve a big problem in less than 24 hours.
I don't know how to thank you yet, but I'll find it.
Trix7777 you are a legend!
thank you very much.MULTi . :)

**Comment by 3v3rm4rt3lL** [comment]
You should hide all personal information when upload files here.
In my opinion, the easy way to do that is:
First all, delete all previous directory m2strm.exe /D
Open with a notepad your original.m3u8 file 
Go to replace and search " FR " and replace for an empy space. And clic to apply to all.
Do the same step with all characters that you wouldnt like to have. 
And then, run m2strm.exe original.m3u8
Please share the feedback.

**Reply by laurent734** [reply]
thank you for your reply
the m3u is a 24h trial :)
if there is no parameter in m2strm, I prefer to use a script with the command sed -i 's / | VM | // g' original.txt.
here is the script and it works!

[m2strm.sh.txt](https://github.com/trix7777/m2strm/files/7794399/m2strm.sh.txt)

it may be useful to other people.
thank you and happy New Year's Eve

**Comment by calexicoca** [comment]
Just to confirm that this is not applicable for VOD, only for Live TV Streams?

**Reply by trix7777** [reply]
m2strm is for VOD, not Live TV Streams.

**Comment by laurent734** [comment]
hi trix7777,
I have a favor to ask of you. :)
Can you add an option so that each strm file is stored in a folder.
I host my strm on my nextcloud server and the opening is much faster when each strm file is stored in a folder that bears the title of the film or series.
thank you again for your work

**Comment by trix7777** [comment]
Closing this thread.
